Sources/MongoMobile/MongoMobile.swift

```diff
@@ -5,31 +5,53 @@ import mongo_embedded
 public struct MongoClientSettings {
     /// the database path to use
     public let dbPath: String
-
+    
     /// public member-wise initializer
     public init(dbPath: String) {
         self.dbPath = dbPath
     }
 }
 
-/// Errors associated with `MongoMobile`.
 public enum MongoMobileError: LocalizedError {
     case invalidClient()
     case invalidInstance(message: String)
     case invalidLibrary()
     case instanceDropError(message: String)
     case cleanupError(message: String)
+}
+
+/// Replica enum of mongo_embedded_v1_error typedef
+private enum MongoEmbeddedV1ErrorCode: Int32 {
+    case reportingError = -2
+    case errorUnknown
+    case success
+    case cannotAllocateMemory
+    case exception
+    case libraryAlreadyInitialized
+    case libraryNotInitialized
+    case invalidLibraryHandle
+    case dbInitializationFailed
+    case invalidDbHandle
+    case hasDbHandlesOpen
+    case dbMaxOpen
+    case dbClientsOpen
+    case invalidClientHandle
+    case reentrancyNotAllowed
+}
 
-    /// a string to be printed out when the error is thrown
+public struct MongoEmbeddedV1Error: LocalizedError {
+    private let statusMessage: String
+    private let error: MongoEmbeddedV1ErrorCode
+    
     public var errorDescription: String? {
-        switch self {
```

Comment:
I believe we still need this original `errorDescription` code up in `MongoMobileError`.